operator/v1/types_ingress.go

```diff
@@ -1037,6 +1039,18 @@ type SyslogLoggingDestinationParameters struct {
 // ContainerLoggingDestinationParameters describes parameters for the Container
 // logging destination type.
 type ContainerLoggingDestinationParameters struct {
+	// maxLength is the maximum length of the log message.
+	//
+	// Valid values are integers in the range 480 to 8192, inclusive.
+	//
+	// When omitted, the default value is 1024.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum=8192
+	// +kubebuilder:validation:Minimum=480
+	// +kubebuilder:default=1024
+	// +optional
+	MaxLength uint32 `json:"maxLength,omitempty"`
```

Comment:
I don't think we support using units as part of our APIs, a regular int32 would be in keeping with convention.

OpenAPI cannot represent the difference between a uint and an int so they are disallowed.

Reply:
`uint32` has been used in the `MaxLength` for syslog in
https://github.com/openshift/api/pull/1439/files#diff-891d2cdcb01603ef2a2c1bfb2f6ef32cf2acf8b1471871a5a6e27308d50050f9R1036
This is the `MaxLength` for side car logging.  Could we keep it for consistency?

Reply:
We tend to avoid repeating mistakes of the past where possible. In theory you could update the `uint32` fields to be `int32` as this won't affect the API schema, it would cause some toil in the consumers of the Go types though, I'd rather we don't add new `uint32`

Reply:
FYI, this comes from an [upstream Kube API convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types)
```
Do not use unsigned integers, due to inconsistent support across languages and libraries. Just validate that the integer is non-negative if that's the case.
```

Reply:
It's confusing that the same conventions say, "Do not use unsigned integers" and also, "All public integer fields MUST use the Go `(u)int32` or Go `(u)int64` types, not `(u)int` (which is ambiguous depending on target platform). Internal types may use `(u)int`."  Both statements date to 2015: https://github.com/kubernetes/community/commit/31832344a1c2b32ba9813f032213944543ac1767 and https://github.com/kubernetes/community/commit/ecc0cc2d5b47258f834a82fda4219767c1b0e3f8 added them, respectively.

Reply:
Fair point, using `int32` would satisfy both statements, using `uint32` would only satisfy one so I'd lean towards that personally